### PR TITLE
Replace dataflow stagingLocation with tempLocation in example module.

### PR DESCRIPTION
**Pull request by peihe**

### CODE REVIEW

**Comment by peihe** [comment]
R: @davorbonaci

File: examples/java/src/main/java/org/apache/beam/examples/WordCount.java [hunk 1]
```diff
@@ -173,17 +171,16 @@ public String apply(KV<String, Long> input) {
...
-     * Returns "gs://${YOUR_STAGING_DIRECTORY}/counts.txt" as the default destination.
+     * Returns "gs://${YOUR_TEMP_DIRECTORY}/counts.txt" as the default destination.
      */
     public static class OutputFactory implements DefaultValueFactory<String> {
       @Override
       public String create(PipelineOptions options) {
-        DataflowPipelineOptions dataflowOptions = options.as(DataflowPipelineOptions.class);
-        if (dataflowOptions.getStagingLocation() != null) {
-          return GcsPath.fromUri(dataflowOptions.getStagingLocation())
+        if (options.getTempLocation() != null) {
```
Comment (davorbonaci) [inline examples/java/src/main/java/org/apache/beam/examples/WordCount.java]: are we sure this gets sets by a runner before this code is executed?

File: examples/java/src/main/java/org/apache/beam/examples/WordCount.java [hunk 2]
```diff
@@ -173,17 +171,16 @@ public String apply(KV<String, Long> input) {
...
+          return GcsPath.fromUri(options.getTempLocation())
```
Comment (davorbonaci) [inline examples/java/src/main/java/org/apache/beam/examples/WordCount.java]: Non great.

**Comment by davorbonaci** [comment]
LGTM, modulo resolution of pending comments.

**Pull request closed without merging**